### ISSUE TITLE
It is not support webpack 4

### ISSUE THREAD

**Issue by Rukeith** [jantimon/html-resource-hints-plugin#10]
This package might be upgrade to support webpack 4.
I have this error message.

```
ERROR in   TypeError: callback is not a function

  - index.js:89
    [blog-website]/[resource-hints-webpack-plugin]/index.js:89:7


  - new Promise


  - Hook.js:35 AsyncSeriesWaterfallHook.lazyCompileHook [as _promise]
    [blog-website]/[tapable]/lib/Hook.js:35:21

  - index.js:627
    [blog-website]/[html-webpack-plugin]/index.js:627:47

  - index.js:166 Promise.resolve.then.then.then.then.then.then.then.result
    [blog-website]/[html-webpack-plugin]/index.js:166:18
```

**Comment by VMBindraban** [comment]
I am getting the error above and sometimes the following error when trying to upgrade to webpack 4:

```
    ERROR in   TypeError: Cannot read property 'options' of undefined

      - index.js:54
        [client]/[resource-hints-webpack-plugin]/index.js:54:60

      - new Promise

      - Hook.js:35 AsyncSeriesWaterfallHook.lazyCompileHook [as _promise]
        [client]/[tapable]/lib/Hook.js:35:21

      - index.js:647
        [client]/[html-webpack-plugin]/index.js:647:47

      - index.js:184 Promise.resolve.then.then.then.then.then.then.then.result
        [client]/[html-webpack-plugin]/index.js:184:18

      - next_tick.js:188 process._tickCallback
        internal/process/next_tick.js:188:7
```

**Comment by jantimon** [comment]
Prepared in #12

**Comment by THernandez03** [comment]
I have a question.

In the [`index.js`](https://github.com/jantimon/resource-hints-webpack-plugin/blob/master/index.js) of the plugin we can see the webpack4 support, but when i do a `npm i -D resource-hints-webpack-plugin` i didn't see the same code.

**Comment by jantimon** [comment]
Published 0.0.2
